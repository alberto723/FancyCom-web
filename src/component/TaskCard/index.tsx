import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
// import './style.css';

export interface TaskCardProps {
  icon?: object;
  title?: string;
  message?: string;
  color?: string;
  href?: string;
  buttonIsVisible?: boolean;
  buttonHref?: string;
  buttonLabel?: string;
  buttonComponent?: object;
}

export const TaskCard: React.FC<TaskCardProps> = ({
  icon,
  title,
  message,
  color,
  href,
  buttonIsVisible,
  buttonHref,
  buttonLabel,
  buttonComponent
}) => {
  const styles: any = useStyles();
  const renderTaskCard = () => {
    return (
      <div
        onClick={() => {
          if (href !== null) {
            window.open(href, '_black');
          }
        }}
        className={styles.container}>
        <div>
          <div className={styles.leftBar} style={{ backgroundColor: color }} />
        </div>
        <div className={styles.taskCard}>
          <div className={styles.iconSection}>{icon}</div>
          <div className={styles.textSection}>
            <p className={styles.titleText}>{title}</p>
            <p className={styles.contentText}>{message}</p>
          </div>
          {buttonIsVisible && buttonComponent == null ? (
            <div
              onClick={() => {
                if (buttonHref !== null) {
                  window.open(buttonHref, '_black');
                }
              }}
              className={styles.buttonSection}
              style={{ backgroundColor: color }}>
              <p className={styles.buttonText}>{buttonLabel}</p>
            </div>
          ) : (
            buttonComponent
          )}
        </div>
      </div>
    );
  };

  return <div className={styles.screen}>{renderTaskCard()}</div>;
};

TaskCard.defaultProps = {
  color: '#81BC7E',
  buttonLabel: 'Button'
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
    marginTop: '15px',
  },
  container: {
    display: 'flex',
    flexDirection: 'row',
    marginLeft: '15px',
    marginRight: '15px',
  },
  leftBar: {
    flex: 1,
    width: '5px',
    height: '100%',
    backgroundColor: '#81BC7E', 
  },
  taskCard: {
    width: '100%',
    padding: '10px',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: '#EBF0F4',
    borderTopWidth: '2px',
    borderBottomWidth: '2px',
    borderRightWidth: '2px',
    borderStyle: 'solid',
  },
  iconSection: {
    marginLeft: '10px',
    marginRight: '10px',
  },
  textSection: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    marginLeft: '10px',
    marginRight: '10px',
  },
  titleText: {
    color: '#425867',
    fontWeight: 'bold',
    fontSize: 14
  },
  contentText: {
    color: '#4F6272',
    fontWeight: 500,
    fontSize: 12,
    marginTop: '5px',
  },
  buttonSection: {
    backgroundColor: '#81BC7E',
    padding: '5px',
    borderRadius: '5px',
    marginLeft: '10px',
    marginRight: '10px',
  },
  buttonText: {
    color: 'white',
    fontWeight: 500,
    fontSize: 12,
  }
}));
