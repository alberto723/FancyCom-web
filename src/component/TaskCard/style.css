.screen {
  flex: 1;
  background-color: white;
  margin-top: 10px;
}
.container {
  display: flex;
  flex-direction: row;
  margin-left: 15px;
  margin-right: 15px;
}
.leftBar {
  flex: 1;
  width: 5px;
  height: 100%;
  background-color: #81BC7E,
}
.taskCard {
  width: 100%;
  padding: 10;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border-color: #EBF0F4;
  border-top-width: 2px;
  border-bottom-width: 2px;
  border-right-width: 2px;
  border-style: solid;
}
.iconSection {
  margin-left: 10px;
  margin-right: 10px;
}
.textSection {
  display: flex;
  flex: 1;
  flex-direction: column;
  margin-left: 10px;
  margin-right: 10px;
}
.titleText {
  color: #425867;
  font-weight: bold;
  font-size: 14;
}
.contentText {
  color: #4F6272;
  font-weight: 500;
  font-size: 12;
  margin-top: 5px;
}
.buttonSection {
  background-color: #81BC7E;
  padding: 5px;
  border-radius: 5px;
  margin-left: 10px;
  margin-right: 10px;
}
.buttonText {
  color: white;
  font-weight: 500;
  font-size: 12px,
}
