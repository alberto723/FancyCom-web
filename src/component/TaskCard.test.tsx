// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { TaskCard } from './TaskCard';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import CloudIcon from '@material-ui/icons/Cloud';
import ExploreIcon from '@material-ui/icons/Explore';

test('Should be rendered in the dom', () => {
  const component = renderer.create(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should show the icon in the icon props in the dom', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-iconSection-5')?.innerHTML != null ).toBe(true);
});

test('Should show a title if one is provided in the title props', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-titleText-7')?.innerHTML != null ).toBe(true);
});

test('Should show a message if one is provided in the message props', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-contentText-8')?.innerHTML != null ).toBe(true);
});

test('Should show the card with the border color that is found in the color prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-taskCard-4')?.innerHTML != null ).toBe(true);
});

test('Should link to the url in the href prop if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-buttonSection-9')?.innerHTML != null ).toBe(true);
});

test('Should show a button if the buttonIsVisible prop is true', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-buttonSection-9')?.innerHTML != null ).toBe(true);
});

test('Should link the button to the url in the buttonHref prop', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-buttonSection-9')?.innerHTML != null ).toBe(true);
});

test('Should render the component in the buttonComponent prop instead of the default button if one is provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <TaskCard
      icon={<CloudIcon style={{ color: '#2C8AC8', fontSize: 30 }} />}
      title='Publish first funnel'
      message='Donec aliquet ipsum vel vehicula'
      color='#2C8AC8'
      href='https://www.google.com'
      buttonIsVisible={true}
      buttonLabel='Cloud'
    />
  );

  expect(container.querySelector('.makeStyles-buttonSection-9')?.innerHTML != null ).toBe(true);
});
