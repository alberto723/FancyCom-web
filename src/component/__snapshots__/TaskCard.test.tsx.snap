// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-container-2"
    onClick={[Function]}
  >
    <div>
      <div
        className="makeStyles-leftBar-3"
        style={
          Object {
            "backgroundColor": "#2C8AC8",
          }
        }
      />
    </div>
    <div
      className="makeStyles-taskCard-4"
    >
      <div
        className="makeStyles-iconSection-5"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "color": "#2C8AC8",
              "fontSize": 30,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96z"
          />
        </svg>
      </div>
      <div
        className="makeStyles-textSection-6"
      >
        <p
          className="makeStyles-titleText-7"
        >
          Publish first funnel
        </p>
        <p
          className="makeStyles-contentText-8"
        >
          Donec aliquet ipsum vel vehicula
        </p>
      </div>
      <div
        className="makeStyles-buttonSection-9"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#2C8AC8",
          }
        }
      >
        <p
          className="makeStyles-buttonText-10"
        >
          Cloud
        </p>
      </div>
    </div>
  </div>
</div>
`;
